framework/src/base/MooseApp.C

```diff
@@ -290,6 +291,21 @@ MooseApp::runInputFile()
   _action_warehouse.executeAllActions();
   _executioner = _action_warehouse.executioner();
 
+  if (getParam<bool>("list_constructed_objects"))
+  {
+    // TODO: ask multiapps for their constructed objects
+    std::vector<std::string> obj_list = _factory.getConstructedObjects();
+    Moose::out << "**START OBJECT DATA**\n";
```

Comment:
I wonder how much pain and suffering this line of code will cause... :smiling_imp: